test/functional/Java19andUp/src/org/openj9/test/jep424/upcall/UpcallMHWithStructTests.java

```diff
@@ -0,0 +1,2906 @@
+/*******************************************************************************
+ * Copyright (c) 2021, 2022 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package org.openj9.test.jep424.upcall;
+
+import org.testng.annotations.Test;
+import org.testng.Assert;
+import org.testng.AssertJUnit;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.VarHandle;
+
+import java.lang.foreign.Addressable;
+import java.lang.foreign.Linker;
+import java.lang.foreign.FunctionDescriptor;
+import java.lang.foreign.GroupLayout;
+import java.lang.foreign.MemoryAddress;
+import java.lang.foreign.MemoryLayout;
+import java.lang.foreign.MemoryLayout.PathElement;
+import java.lang.foreign.MemorySegment;
+import java.lang.foreign.MemorySession;
+import java.lang.foreign.SegmentAllocator;
+import java.lang.foreign.SequenceLayout;
+import java.lang.foreign.SymbolLookup;
+import java.lang.foreign.ValueLayout;
+import static java.lang.foreign.ValueLayout.*;
+
+/**
+ * Test cases for JEP 424: Foreign Linker API (Preview) for argument/return struct in upcall.
+ *
+ * Note: the padding elements in the struct are only required by RI or VarHandle (accessing the
+ * data address) while they are totally ignored in OpenJ9 given the padding/alignment are
+ * computed by libffi automatically in native.
+ */
+@Test(groups = { "level.sanity" })
+public class UpcallMHWithStructTests {
+	private static String osName = System.getProperty("os.name").toLowerCase();
+	private static String arch = System.getProperty("os.arch").toLowerCase();
+	private static boolean isLinuxS390x = osName.contains("linux") && arch.equals("s390x");
```

Comment:
Windows does not require the structure padding either, so all places that check for `isLinuxS390x` will have to check for Windows as well.

I wonder if it would be better to introduce a `requiresStructurePadding` boolean here that is initialized to false for Z and Windows rather than adding OS checks for many of the tests.  It seems a bit cleaner in my opinion.

Same for all other versions of UpcallMHWithStructTests.java.

Reply:
I will update the test on Java17/18/19 based against the suggestion above.